ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{deneb::blob_sidecar::Blob, primitives::Bytes32, ssz::prelude::*};
+use c_kzg::Error;
+pub use c_kzg::KzgSettings;
+
+use super::mainnet::FIELD_ELEMENTS_PER_BLOB;
+
+pub const BYTES_PER_FIELD_ELEMENT: usize = 32;
+pub const BYTES_PER_BLOB: usize = BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB;
+pub const BYTES_PER_COMMITMENT: usize = 48;
+pub const BYTES_PER_PROOF: usize = 48;
+
+pub type Bytes48 = ByteVector<48>;
+pub type VersionedHash = Bytes32;
+pub type FieldElement = Bytes32;
+pub type KzgCommitment = Bytes48;
+pub type KzgProof = Bytes48;
+
+pub struct ProofAndEvaluation {
+    pub proof: KzgProof,
+    pub evaluation: Bytes32,
+}
+
+pub fn blob_to_kzg_commitment<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgCommitment, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref()).unwrap();
+
+    let commitment = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).unwrap();
+    Ok(inner)
+}
+
+pub fn compute_kzg_proof<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    evaluation_point: &FieldElement,
+    kzg_settings: &KzgSettings,
+) -> Result<ProofAndEvaluation, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref()).unwrap();
+    let evaluation_point = c_kzg::Bytes32::from_bytes(evaluation_point.as_ref())?;
+
+    let (proof, evaluation) =
+        c_kzg::KzgProof::compute_kzg_proof(&blob, &evaluation_point, kzg_settings)?;
+    let proof = ByteVector::try_from(proof.to_bytes().as_ref()).unwrap();
+    let evaluation = ByteVector::try_from(evaluation.as_slice()).unwrap();
+
+    let result = ProofAndEvaluation { proof, evaluation };
+    Ok(result)
+}
+
+pub fn compute_blob_kzg_proof<const BYTES_PER_BLOB: usize>(
+    blob: &Blob<BYTES_PER_BLOB>,
+    commitment: &KzgCommitment,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgProof, Error> {
+    let blob = c_kzg::Blob::from_bytes(blob.as_ref()).unwrap();
+    let commitment = c_kzg::Bytes48::from_bytes(commitment.as_ref()).unwrap();
+
+    let ckzg_proof = c_kzg::KzgProof::compute_blob_kzg_proof(&blob, &commitment, kzg_settings)?;
+
+    let bytes_proof = ckzg_proof.to_bytes();
+    let proof = ByteVector::try_from(bytes_proof.as_ref()).unwrap();
+    Ok(proof)
```

Comment:
this is more stylistic but we can be a bit more compact

```rust
    let proof = c_kzg::KzgProof::compute_blob_kzg_proof(&blob, &commitment, kzg_settings)?;
    Ok(ByteVector::try_from(proof.to_bytes().as_ref()).unwrap())
```

another thing is that we aren't using our type alias we went to all the trouble to define, so I'd argue the intent of this code is clearer if we do this

```rust
    let proof = c_kzg::KzgProof::compute_blob_kzg_proof(&blob, &commitment, kzg_settings)?;
    Ok(KzgProof::try_from(proof.to_bytes().as_ref()).unwrap())
```

and we can actually go one step further using `TryInto` and some type inference

```rust
Ok(proof.to_bytes().try_into().expect("input is correct size"))
```

Reply:
Idk how to implement From for it yet.  Sticking with ` Ok(KzgProof::try_from(proof.to_bytes().as_ref()).expect("input is correct size"))` for now